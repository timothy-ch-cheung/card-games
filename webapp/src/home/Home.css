.button-panel {
    padding: 10px;
    margin: 10px auto;
    width: 70%;
    display: table;
    text-align: center;
}

h1 {
    text-align: center;
    padding: 10px;
}

.container {
    margin-top: 20px;
    margin-bottom: 40px;
}