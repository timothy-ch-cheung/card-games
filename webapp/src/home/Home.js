import React from "react";
import ButtonGroup from "react-bootstrap/ButtonGroup";
import Button from "react-bootstrap/Button";
import './Home.css';
import {LinkContainer} from 'react-router-bootstrap'
import Board from "../components/Board";
import CreateGame from "../components/CreateGame";

class Home extends React.Component {

    state = {
        show: false
    };

    handleShow = () => {
        this.setState({show: true});
    }

    handleHide = () => {
        this.setState({show: false});
    }

    render() {
        return (
            <div>
                <style type="text/css">
                    {`
                        .btn-xl {
                            font-size: 1.5rem;
                            border: grey solid 1px;
                            width: 200px;
                        }
                        
                        @media screen and (max-width: 768px) {
                            .btn-xl {
                                font-size: 1.2rem;
                                width: 150px;
                            }
                        }
                    `}
                </style>
                <h1>Play against a Bot:</h1>
                <Board/>
                <h1>Play with others:</h1>
                <div className="button-panel">
                    <ButtonGroup aria-label="Basic example">
                        <Button size="xl" variant="info" onClick={e => {
                            this.handleShow();
                        }}>Create Game</Button>
                        <CreateGame show={this.state.show} onClose={this.handleHide}/>
                        <LinkContainer to="/games">
                            <Button size="xl" variant="light">Join Game</Button>
                        </LinkContainer>
                    </ButtonGroup>
                </div>
            </div>
        )
    }
}

export default Home