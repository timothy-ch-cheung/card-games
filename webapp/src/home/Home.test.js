import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from "./Home";

configure({ adapter: new Adapter() })
test('matches snapshot', () => {
    const tree = shallow(<Home/>);
    expect(tree).toMatchSnapshot()
});