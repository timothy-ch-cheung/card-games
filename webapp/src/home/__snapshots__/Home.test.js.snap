// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Home snapshot 1`] = `
<div>
  <style
    type="text/css"
  >
    
                        .btn-xl {
                            font-size: 1.5rem;
                            border: grey solid 1px;
                            width: 200px;
                        }
                        
                        @media screen and (max-width: 768px) {
                            .btn-xl {
                                font-size: 1.2rem;
                                width: 150px;
                            }
                        }                        
                    
  </style>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
        "marginBottom": "10px",
        "marginTop": "10px",
        "width": "100%",
      }
    }
  >
    <Image
      className="no-drag"
      fluid={true}
      rounded={false}
      roundedCircle={false}
      src="card-banner.svg"
      style={
        Object {
          "width": "70%",
        }
      }
      thumbnail={false}
    />
  </div>
  <h1>
    PLAY:
  </h1>
  <div
    className="button-panel"
  >
    <ButtonGroup
      aria-label="Basic example"
      role="group"
      toggle={false}
      vertical={false}
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        size="xl"
        type="button"
        variant="info"
      >
        Create Game
      </Button>
      <CreateGame
        onClose={[Function]}
        show={false}
      />
      <withRouter(LinkContainer)
        to="/games"
      >
        <Button
          active={false}
          disabled={false}
          size="xl"
          type="button"
          variant="light"
        >
          Join Game
        </Button>
      </withRouter(LinkContainer)>
    </ButtonGroup>
  </div>
</div>
`;
